sigopt/xgboost/run.py

```diff
@@ -84,14 +86,21 @@ def parse_run_options(run_options):
     if 'run' in run_options.keys() and run_options['run'] is not None:
       if not isinstance(run_options['run'], RunContext):
         raise TypeError(
-          "`run` must be an instance of RunContext object, not {type(run_options['run']).__name__}."
+          f"`run` must be an instance of RunContext object, not {type(run_options['run']).__name__}."
         )
 
     return {**DEFAULT_RUN_OPTIONS, **run_options}
 
   return copy.deepcopy(DEFAULT_RUN_OPTIONS)
 
 
+def validate_xgboost_kwargs(xgb_kwargs):
+  if xgb_kwargs:
+    for key in list(xgb_kwargs.keys()):
+      if key not in signature(xgboost.train).parameters.keys():
+        xgb_kwargs.pop(key)
```

Comment:
I'm fine removing the key, but perhaps we should at least throw a warning to tell the user what we are doing:

```
import warnings 
...
  warnings.warn(f"The argument {key} is not supported by XGBoost, and has been ignored" )
```

Reply:
sounds good. Will use `RuntimeWarning` class.

Reply:
I guess one potential issue is that there may be keys we want to pass in as kwargs that aren't actually directly passed into the integration, but that's not the situation now and we can address this down the road if needed